CONTRIBUTING.md

```diff
@@ -182,50 +182,79 @@ are stored under `_build/html`. The structure of `_build/html` is as follows:
-    + [agda-sets](https://github.com/input-output-hk/agda-sets)
++  In a folder `LIB`, clone the dependencies
+   + [agda-stdlib](https://github.com/agda/agda-stdlib)
+   + [agda-stdlib-classes](https://github.com/agda/agda-stdlib-classes)
+   + [agda-stdlib-meta](https://github.com/agda/agda-stdlib-meta)
+   + [agda-sets](https://github.com/input-output-hk/agda-sets)
 
-and checkout the commits/tags found in `default.nix` (e.g. `v2.1.1` for `agda-stdlib-meta`).
+   ```
+   mkdir -p LIB; cd LIB
+   git clone --config advice.detachedHead=false --single-branch \
+     -b "v2.2" https://github.com/agda/agda-stdlib.git
+   git clone --config advice.detachedHead=false --single-branch \
+     -b "v2.0" https://github.com/agda/agda-stdlib-classes.git
+   git clone --config advice.detachedHead=false --single-branch \
+     -b "v2.1.1" https://github.com/agda/agda-stdlib-meta.git
+   git clone --config advice.detachedHead=false --single-branch \
+     -b "master" https://github.com/input-output-hk/agda-sets.git
+   ```
 
-- Create a file `LIB/libraries` with the following content:
-```
-LIB/agda-stdlib/standard-library.agda-lib
-LIB/agda-stdlib-classes/agda-stdlib-classes.agda-lib
-LIB/agda-stdlib-meta/agda-stdlib-meta.agda-lib
-LIB/agda-sets/abstract-set-theory.agda-lib
-```
++  Create a file `LIB/libraries` with the following content:
 
-- Use `AGDA_DIR=LIB agda` instead of `agda`.
-  For example:
-    - To typecheck the formal specification, run:
-    ```
-    AGDA_DIR=LIB agda src/Everything.agda
-    ```
+   ```
+   LIB/agda-stdlib/standard-library.agda-lib
+   LIB/agda-stdlib-classes/agda-stdlib-classes.agda-lib
+   LIB/agda-stdlib-meta/agda-stdlib-meta.agda-lib
+   LIB/agda-sets/abstract-set-theory.agda-lib
+   ```
 
-    - To build the `conway-ledger.pdf` artifact, run:
-    ```
-    AGDA_DIR=LIB fls-shake conway-ledger.pdf
-    ```
++  Use `AGDA_DIR=LIB agda` instead of `agda`.
+
+   For example:
+
+    +  To typecheck the formal specification, run:
+
+       ```
+       AGDA_DIR=LIB agda src/Everything.agda
+       ```
+
+    +  To build the `conway-ledger.pdf` artifact, run:
+
+       ```
+       AGDA_DIR=LIB fls-shake conway-ledger.pdf
+       ```
+
+---
 
 ### `fls-shake`
 
-`fls-shake` can be compiled the file
-[`Shakefile.hs`](Shakefile.hs) using GHC.
+`fls-shake` can be compiled from the file [`Shakefile.hs`](Shakefile.hs) using GHC.
+
+For this, do the same as the derivation `fls-shake` in [default.nix](default.nix):
+
++  Install the packages it depends upon, which are listed under `nativeBuildInputs`
 
-For this, do the same as the derivation `fls-shake` in
-[default.nix](default.nix):
+   haskellPackages.ghcWithPackages (ps: with ps;
+                            ([ shake binary deepseq hashable ]))) ];
+    buildPhase = ''
+      ghc -o fls-shake Shakefile.hs -threaded
+    '';
+    installPhase = ''
+      mkdir -p "$out/bin"
+      cp fls-shake "$out/bin/"
+    '';
+  };
```

Comment:
I didn't include the packages here on purpose so they do not become out of  sync in case something changes.